src/libs/actions/Transaction.js

```diff
@@ -31,4 +52,49 @@ function addStop(transactionID, newLastIndex) {
     });
 }
 
-export {addStop, createInitialWaypoints};
+/**
+ * Saves the selected waypoint to the transaction
+ * @param {String} transactionID
+ * @param {String} index
+ * @param {Object} waypoint
+ */
+function saveWaypoint(transactionID, index, waypoint) {
+    Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`, {
+        comment: {
+            waypoints: {
+                [`waypoint${index}`]: waypoint,
+            },
+        },
+    });
+}
+
+function removeWaypoint(transactionID, currentIndex) {
+    // Index comes from the route params and is a string
+    const index = Number(currentIndex);
+    const transaction = lodashGet(allTransactions, transactionID, {});
+    const existingWaypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const totalWaypoints = _.size(existingWaypoints);
+
+    // Prevents removing the starting or ending waypoint but clear the stored address
+    if (index === 0 || index === totalWaypoints - 1) {
+        saveWaypoint(transactionID, index, null);
+        return;
+    }
+
+    const waypointValues = _.values(existingWaypoints);
+    waypointValues.splice(index, 1);
+
+    const reIndexedWaypoints = {};
+    waypointValues.forEach((waypoint, idx) => {
+        reIndexedWaypoints[`waypoint${idx}`] = waypoint;
+    });
+
+    // Onyx.merge won't remove the null nested object values, this is a workaround
+    // to remove nested keys while also preserving other object keys
```

Comment:
NAB: I like this comment thanks for adding the context. This is a pain. Would you please see if there's an existing Onyx issues and if not create one to remove the null nested object values?

Reply:
https://github.com/Expensify/react-native-onyx/issues/301